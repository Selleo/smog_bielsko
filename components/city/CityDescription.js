import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default class CityDescription extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <View>
        <Text style={styles.text}>
          {this.props.text}
        </Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  text: {
    fontSize: 14
  }
})
