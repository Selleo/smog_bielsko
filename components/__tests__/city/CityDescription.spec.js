import 'react-native';
import React                          from 'react';

import { shallow }                    from 'enzyme';
import chai, { expect }               from 'chai';
import dirtyChai                      from 'dirty-chai';

import CityDescription                from '../../city/CityDescription';

chai.use(dirtyChai);

let wrapper, props;

describe('CityDescription', () => {
  it('renders correctly', () => {
    props = {text: 'ul. Rynek 2'};
    wrapper = shallow(<CityDescription {...props}/>);

    expect(wrapper).to.exist();
  });

  it('renders description on html', () => {
    props = {text: 'ul. Rynek 2'};
    wrapper = shallow(<CityDescription {...props}/>);
    expect(wrapper.children().length).to.equal(1);

    expect(wrapper.children().children().text()).to.equal('ul. Rynek 2');
  });

});

