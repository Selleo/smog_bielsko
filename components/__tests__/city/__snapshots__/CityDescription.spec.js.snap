exports[`CityDescription renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "fontSize": 14,
              }
        }>
        ul. Rynek 2
    </Text>
</View>,
  "nodes": Array [
    <View>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                  Object {
                        "fontSize": 14,
                      }
            }>
            ul. Rynek 2
      </Text>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CityDescription
        text="ul. Rynek 2" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CityDescription {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "text": "ul. Rynek 2",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "fontSize": 14,
                                      }
                    }>
                    ul. Rynek 2
          </Text>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "fontSize": 14,
                                      }
                    }>
                    ul. Rynek 2
          </Text>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CityDescription
    text="ul. Rynek 2" />,
}
`;
